### PR TITLE
refac: No longer use `OperationType` for charges

### PR DESCRIPTION
**Pull request by prtandrup**
## Description

If applied, this PR refactors the code base to no longer use `OperationType` for `ChargeCommand.cs` and `Charge.cs` 

As OperationType with the values "Create, Update and Delete" is an ebIX origin, that does not fly with CIM format, it will be removed. Later PR's will ensure that an incoming charge will update any stored data for the same Charge correctly. 

## References

* #288

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Factories/ChargeFactory.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public Task<Charge> CreateFromCommandAsync([NotNull]ChargeCommand command)
                 TransparentInvoicing = command.ChargeOperation.TransparentInvoicing,
                 EndDateTime = command.ChargeOperation.EndDateTime,
                 StartDateTime = command.ChargeOperation.StartDateTime,
-                Status = command.ChargeOperation.OperationType,
+                Status = OperationType.Create,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Factories/ChargeFactory.cs:41]: For now this was done to ensure passing unit tests. I will remove the use of Status in Charge.cs in a future commit.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ChargeCommandInputValidator.cs [hunk 2]
```diff
@@ -29,25 +29,8 @@ public ChargeCommandInputValidator(IInputValidationRulesFactory inputValidationR
...
-                    return ruleSet.Validate();
-                case OperationType.Update:
-                    ruleSet = _inputValidationRulesFactory.CreateRulesForChargeUpdateCommand(chargeCommand);
-                    return ruleSet.Validate();
-                default:
-                    ruleSet = _inputValidationRulesFactory.CreateRulesForChargeUnknownCommand(chargeCommand);
-                    return ruleSet.Validate();
-            }
+            IValidationRuleSet ruleSet = _inputValidationRulesFactory.CreateRulesForChargeCommand(chargeCommand);
+            return ruleSet.Validate();
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ChargeCommandInputValidator.cs:33]: This class has no unit tests. Should we add some?

Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ChargeCommandInputValidator.cs:33]: added

**Review by bemwen — APPROVED** [review]
LGTM, well done :)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessValidationRulesFactoryTests.cs [hunk 3]
```diff
@@ -30,51 +36,97 @@ namespace GreenEnergyHub.Charges.Tests.Application.Validation.BusinessValidation
...
+                    It.IsAny<string>(),
+                    It.IsAny<ChargeType>()))
+                .Returns(Task.FromResult(charge!));
+
+            // Act
+            var actual = await sut.CreateRulesForChargeCommandAsync(chargeCommand).ConfigureAwait(false);
+            var actualRules = actual.GetRules().Select(r => r.GetType());
+
+            // Assert
+            Assert.Equal(2, actual.GetRules().Count);
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessValidationRulesFactoryTests.cs:64]: why is the amount of rules important for the test?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessValidationRulesFactoryTests.cs [hunk 4]
```diff
@@ -30,51 +36,97 @@ namespace GreenEnergyHub.Charges.Tests.Application.Validation.BusinessValidation
...
+            Assert.Contains(expectedRule, actualRules);
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessValidationRulesFactoryTests.cs:65]: How does this work? 
expectedRule is a 'Type' so would this not always be present in a collection of types?

**Review by lasrinnil — APPROVED** [review]
Good job! Just a few questions 👍

**Pull request merged**